### PR TITLE
700 audit log

### PR DESCRIPTION
**Pull request by michaelkruglos**
Implementation of audit logging

### CODE REVIEW

File: services/secure-gateway/audit/logAudit.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	log *log.Logger
+}
+
+// New creates a new Auditor, which logs to io.Writer, with prefix 'AUDIT'
+func New(out io.Writer) (Auditor, error) {
+	return NewLog(log.New(out, "AUDIT", log.LstdFlags|log.LUTC))
+}
+
+// NewLog creates a new logger based Auditor
+func NewLog(logger *log.Logger) (Auditor, error) {
```
Comment (AleF83) [inline services/secure-gateway/audit/logAudit.go]: Consider naming it NewLogger

File: services/secure-gateway/audit/interface.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package audit
```
Comment (AleF83) [inline services/secure-gateway/audit/interface.go]: Consider naming it Auditor.go

File: services/secure-gateway/main.go [hunk 3]
```diff
@@ -52,12 +55,29 @@ func NewApp(configuration *config.Configuration, token security.JWTToken) http.H
...
 	enforcer.EnableEnforce(configuration.Security.Enforce)
 
-	router := NewRouter(configuration)
-	authenticationMiddleware := security.AuthenticationMiddleware(configuration.Security)
-	authorizationMiddleware := security.AuthorizationMiddleware(enforcer)
+	auditor, err := audit.New(os.Stdout)
+	if err != nil {
+		panic("Unable to create security auditing log")
+	}
+	if configuration.Security.Enforce {
```
Comment (AleF83) [inline services/secure-gateway/main.go]: Consider one function like LogEnforserStatus that recieves configuration and logs the state according to it

File: services/secure-gateway/main.go [hunk 4]
```diff
@@ -52,12 +55,29 @@ func NewApp(configuration *config.Configuration, token security.JWTToken) http.H
...
+		auditor.EnforcerEnabled()
+	} else {
+		auditor.EnforcerDisabled()
+	}
+
+	var authenticationMiddleware negroni.HandlerFunc
+	if _, testing := os.LookupEnv("TWEEK_TESTING"); testing {
```
Comment (AleF83) [inline services/secure-gateway/main.go]: This flag doesn't exist any more

File: services/secure-gateway/security/authentication.go [hunk 5]
```diff
@@ -88,6 +90,16 @@ func AuthenticationMiddleware(configuration *config.Security) negroni.HandlerFun
 	})
 }
 
+// TestingAuthenticationMiddleware enriches the request's context with the user info for testing
```
Comment (AleF83) [inline services/secure-gateway/security/authentication.go]: No need any more

File: services/secure-gateway/main.go (hunk 3 above)
Comment (michaelkruglos) [reply services/secure-gateway/main.go]: We'll keep it as is for now, if we encounter problems with it - we'll fix it.

**Pull request merged**